### PR TITLE
Adding typescript support 

### PR DESCRIPTION
**Pull request by ltphen**
I have written an index.d.ts file for the project in order to use it with typescript because it is not compatible with typescript actually

### CODE REVIEW

**Comment by confuser** [comment]
Thanks for this. I'll merge once I've migrated CI over to github actions

**Comment by vespaiach** [comment]
@ltphen I think the `package.json` file  should be updated as well.

```
{
  ...
  "types": "index.d.ts"
}
```

Refer to [https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import {GraphQLSchema} from "graphql";
+
+export function constraintDirective () : (schema: GraphQLSchema) => GraphQLSchema;
+
+export const constraintDirectiveTypeDefs: String
```
Comment (karavaan) [inline index.d.ts:5]: `String` should be `string`

Comment (confuser) [reply index.d.ts:5]: @ltphen will merge this in once the comments have been addressed. Thanks!

Comment (ltphen) [reply index.d.ts]: Hey @confuser, The changes has been taken into account

**Pull request merged**